### PR TITLE
CLN: unify error message when parsing datetimes with mixed time zones with utc=False

### PR DESCRIPTION
**Pull request by natmokval**
xref #57275, #55793

unified ValueError message when parsing datetimes with mixed time zones with utc=False
so far we get two different ValueError messages:
```
import pandas as pd
from datetime import datetime
import pytz

str_my ="2020-10-26 00:00:00+06:00"
ts = pd.Timestamp("2018-01-01", tz="US/Pacific")
dt = datetime(2020, 1, 1, 18, tzinfo=pytz.timezone("Australia/Melbourne"))

dt1 = pd.to_datetime([ts, str_my])
# ValueError: Mixed timezones detected. pass utc=True in to_datetime or tz='UTC' in DatetimeIndex to convert to a common timezone.
dt2 = pd.to_datetime([str_my, ts])
# ValueError: cannot parse datetimes with mixed time zones unless `utc=True`

dt1 = pd.to_datetime([dt, str_my], utc=False)
# ValueError: Mixed timezones detected. pass utc=True in to_datetime or tz='UTC' in DatetimeIndex to convert to a common timezone.
dt2 = pd.to_datetime([str_my, dt], utc=False)
# ValueError: cannot parse datetimes with mixed time zones unless `utc=True`
```

### CODE REVIEW

File: pandas/_libs/tslib.pyx [hunk 1]
```diff
@@ -598,7 +598,8 @@ cpdef array_to_datetime(
         is_same_offsets = len(out_tzoffset_vals) == 1
         if not is_same_offsets:
             raise ValueError(
-                "cannot parse datetimes with mixed time zones unless `utc=True`"
+                "Mixed timezones detected. pass utc=True in to_datetime "
```
Comment (mroeschke) [inline pandas/_libs/tslib.pyx:601]:
```suggestion
                "Mixed timezones detected. Pass utc=True in to_datetime "
```

And could you capitalize elsewhere too?

**Review by MarcoGorelli — APPROVED** [review]
nice one, thanks!

File: pandas/_libs/tslib.pyx (hunk 1 above)
Comment (natmokval) [reply pandas/_libs/tslib.pyx]: Thank you for reviewing this PR. I capitalized the letter 'p'.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @natmokval

**Pull request merged**